cuda/cuda_nemolite2d.cu

```diff
@@ -4,12 +4,38 @@
 
 #include <cuda_runtime.h>
 
+#include "cuda_utils.cuh"
 #include "fortran_array_2d.cuh"
 
 // Working precision
 typedef double wp_t;
 
-// <<<number_of_blocks, size_of_block>>>
+inline void
+get_kernel_dims(const int max_x,
+                const int max_y,
+                dim3& out_blocksize,
+                dim3& out_gridsize)
+{
+
+  // Suggest block dimensions. Threads per block must not exceed 1024 on most
+  // hardware, registers will probably be a limiting factor.
+  dim3 blocksize(2, 2);
```

Comment:
2x2 is too small for good performance ( atleast 32 threads will be launched). 

Additionally, different kernels use differnt numbers of registers, so the "best" blocksize may be different per kernel. This can either be found through trial and error / looking at profiler output and optimising for occupancy. The occupancy calculator api is an option, though it's unclear how it works for 2D launch. I believe you tell it the total number fo threads you want and it returns the number of threads per block (which you are able to divide into 2D as you wish)